libs/git/clone.go

```diff
@@ -70,3 +62,29 @@ func Clone(ctx context.Context, url, reference, targetPath string) error {
 	}
 	return nil
 }
+
+func Clone(ctx context.Context, url, reference, targetPath string) error {
+	// We assume only the repository name has been if input does not contain any
+	// `/` characters and the url is only made up of alphanumeric characters and
+	// ".", "_" and "-". This repository is resolved again databricks github account.
+	fullUrl := url
+	if githubRepoRegex.MatchString(url) {
+		fullUrl = strings.Join([]string{githubUrl, databricksOrg, url}, "/")
+	}
+
+	opts := cloneOptions{
+		Reference:     reference,
+		RepositoryUrl: fullUrl,
+		TargetPath:    targetPath,
+		Shallow:       true,
+	}
+
+	err := opts.clone(ctx)
+	// Git repos hosted via HTTP do not support shallow cloning. We try with
+	// a deep clone this time
+	if err != nil && strings.Contains(err.Error(), "dumb http transport does not support shallow capabilities") {
+		opts.Shallow = false
+		err = opts.clone(ctx)
```

Comment:
Nit: you can `return` this directly as anything happening below will be irrelevant.